OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -63,6 +63,8 @@ public class ArmamentInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		public readonly bool ShootAtClosestTargetablePosition = false;
```

Comment:
fwiw, I'm in favor of that, too.

Now, for RA/TD there are two options:
a) leave health shapes as they are and just make buildings CenterPosition-targeted by all logic. This will have less balancing impact, it just reduces the effective weapon range vs. >1-cell structures. The only potential regression would be that some melee units (ants, zombies, dinos) might be unable to attack larger structures. This could be worked around to some degree, but not perfectly. Irrelevant for MP balance though, would only affect a few missions.

b) Get rid of these awful "one-size-fits-all" 10-pixel health radii and give buildings proper cell-based hit shapes in all mods.
Would be more realistic, true to the originals and avoid regressions, but would turn balance of inaccurate weapons (arty, cruiser, parabombs etc.) upside-down.
We also need to support to optionally make `=` occupied-but-passable cells be included in the hit shape and targetable positions (for repair bays, mostly), but that should be relatively trivial.

I'm in favor of b).